### PR TITLE
CSS: Correct misrepresentation of "auto" horizontal margins as 0

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-2237

But as expected, it's pretty big:
```
   raw     gz Compared to compat @ fb25bacf9b809a08c29e76decd8cda1579fdf192    
  +790   +196 dist/jquery.js                                                   
  +290    +78 dist/jquery.min.js
```

### CODE REVIEW

File: src/css/support.js [hunk 1]
```diff
@@ -76,6 +76,14 @@ define([
 				computeStyleTests();
 			}
 			return reliableMarginRightVal;
+		},
+
+		reliableMarginLeft: function() {
+			// Support: IE <=8 only, Android 4.0 - 4.3 only, Firefox <=3 - 37
```
Comment (markelog) [inline src/css/support.js]: Usually, there is no spaces in ranges :-)

Comment (gibson042) [inline src/css/support.js]: jquery/contribute.jquery.org/issues/95

Comment (markelog) [inline src/css/support.js]: Yeah, it seems we didn't had consensus about this issue, it did come down to like/don't like, i'd say we should follow some standart here, if we so nit-picky :-).

http://practicaltypography.com/hyphens-and-dashes.html 
http://en.wikipedia.org/wiki/Dash#Ranges_of_values

**Comment by markelog** [comment]
If consumer reset style it is Firefox only, right? Should we consider asking ff team to prioritize their ticket you mentioned in affiliated issue instead?

**Comment by gibson042** [comment]
I'm not sure I understand your question... this affects [IE8, Android, and Firefox](https://github.com/jquery/jquery/pull/2276/files#diff-7a031a0a341e84edbc03348dea967f7cR81), but I guess it's true that Firefox is the only browser in which the behavior has yet to be fixed.

I probably should have already landed this, but since it's still open, it might as well still be open for discussion.

**Comment by markelog** [comment]
I was referencing <a href="https://github.com/jquery/jquery/issues/2237#issuecomment-97281378">this</a> discussion.

**Comment by timmywil** [comment]
@gibson042 Is this high enough priority to include in our 1.x release?

**Comment by dmethvin** [comment]
Looks like IE8 started failing when this landed: http://swarm.jquery.org/result/720927

**Pull request closed without merging**